addons/mass_mailing/models/mailing.py

```diff
@@ -918,20 +918,26 @@ def _process_mass_mailing_queue(self):
                     # send the KPI mail only if it's the first sending
                     'kpi_mail_required': not mass_mailing.sent_date,
                 })
-
-        mailings = self.env['mailing.mailing'].search([
-            ('kpi_mail_required', '=', True),
-            ('state', '=', 'done'),
-            ('sent_date', '<=', fields.Datetime.now() - relativedelta(days=1)),
-            ('sent_date', '>=', fields.Datetime.now() - relativedelta(days=5)),
-        ])
-        if mailings:
-            mailings._action_send_statistics()
+        if not self.env['ir.config_parameter'].sudo().get_param('mass_mailing.no_mass_mailing_reports'):
+            mailings = self.env['mailing.mailing'].search([
+                ('kpi_mail_required', '=', True),
+                ('state', '=', 'done'),
+                ('sent_date', '<=', fields.Datetime.now() - relativedelta(days=1)),
+                ('sent_date', '>=', fields.Datetime.now() - relativedelta(days=5)),
+            ])
+            if mailings:
+                mailings._action_send_statistics()
 
     # ------------------------------------------------------
     # STATISTICS
     # ------------------------------------------------------
 
+    def _get_unsubscribe_token(self, user_id):
+        """Generate a secure hash for this user. It allows to
+        opt out from mailing reports while keeping some security in that process.
+        """
+        return tools.hmac(self.env(su=True), 'mass_mailing-report-unsubscribe', user_id)
```

Comment:
mailing-report-deactivated maybe ? Just to keep coherency with parameter name (whatever final naming, better trying to keep the same naming style to ease understanding)?.